src/main/perf/SearchPerfTest.java

```diff
@@ -190,9 +195,16 @@ private static void _main(String[] clArgs) throws Exception {
     final String fieldName = args.getString("-field");
     final boolean printHeap = args.getFlag("-printHeap");
     final boolean doPKLookup = args.getFlag("-pk");
+    final boolean doConcurrentSegmentReads = args.getFlag("-concurrentSegReads");
     final int topN = args.getInt("-topN");
     final boolean doStoredLoads = args.getFlag("-loadStoredFields");
 
+    int cores = Runtime.getRuntime().availableProcessors();
+    ExecutorService tempExecutorService = new ThreadPoolExecutor(cores, cores, 0L, TimeUnit.MILLISECONDS,
```

Comment:
Can we move this inside an `if (doConcurrentSearcher) { ... }` so we don't create the `ExecutorService` unless we will use it?